main.go

```diff
@@ -287,10 +287,34 @@ func main() {
 	}
 
 	if config.Window {
-		windowControls := svg.NewWindowControls(5.5*float64(scale), 19.0*scale, 12.0*scale)
+		x := 19.0 * scale
+		y := 12.0 * scale
+		windowControls, controlsWidth := svg.NewWindowControls(5.5*float64(scale), x, y)
 		svg.Move(windowControls, float64(config.Margin[left]), float64(config.Margin[top]))
 		image.AddChild(windowControls)
 		config.Padding[top] += (15 * scale)
+		if config.Title.Text != "" {
+			titleText := config.Title.Text
+			if config.Title.Text == "auto" {
+				titleText = filepath.Base(config.Input)
+			}
+			my := float64(config.Margin[top]) + y
+			ml := float64(config.Margin[left]) + x + controlsWidth
+			mr := float64(config.Margin[right]) + x
+			// [0] left, [1] center, [2] right, [3] top
+			titlePos := []float64{ml, imageWidth / 2, imageWidth - mr, my}
```

Comment:
You should define a struct with 4 fields, it would be clearer, because here you are defining 0-3 then you use 0-3 as magical number in the function where you use it.

Can't you move the computing out the function?

Reply:
It's up to you, whatever makes sense when you code it.